src/app/context/CurrencyContext.tsx

```diff
@@ -0,0 +1,310 @@
+import { useState, useEffect, createContext, useContext } from "react";
+// https://www.npmjs.com/package/currency.js 0 dependencies, 135,987k weekly downloads
+import currencyJs from "currency.js";
+// https://github.com/MikeMcl/decimal.js-light 0 dependencies, 826,917 weekly downloads
+import { Decimal } from "decimal.js-light";
+import axios from "axios";
+
+import { useAuth } from "./AuthContext";
+import api from "~/common/lib/api";
+import { SupportedCurrencies, SupportedExchanges } from "~/types";
+
+interface CurrencyContextType {
+  balances: {
+    satsBalance: string;
+    fiatBalance: string;
+  };
+  setCurrencyValue: (currency: SupportedCurrencies) => void;
+  currencies: string[];
+  getFiatValue: (amount: number | string) => Promise<string>;
+}
+
+const CurrencyContext = createContext({} as CurrencyContextType);
+
+const numSatsInBtc = 100_000_000;
+
+export function CurrencyProvider({ children }: { children: React.ReactNode }) {
+  const currencies = [
```

Comment:
Would the type here be supportedCurrencies? Could we utilise [the existing type ](https://github.com/getAlby/lightning-browser-extension/pull/890/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR211)to create this instead of duplicating it?

Reply:
I'm not sure how to use a type as an array of values. Maybe I'm misunderstanding.

Reply:
Uhm, sorry, maybe the other way around? Somethign liek this?:  
https://stackoverflow.com/a/54612222/1667461  

Just woudl be nice to avoid mainting two times the same list.